ui/src/components/Header/NavBack.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import styled from "styled-components";
+import { Link, useHistory } from "react-router-dom";
+
+const BackLink = styled(Link)`
+  width: 3.5rem;
+  height: 3.5rem;
+  padding: 0;
+  position: absolute;
+  left: 0;
+  background: transparent;
+  border: none;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const NavBack = () => {
+  const {
+    location: { pathname },
+  } = useHistory();
+  const split = pathname.split("/");
```

Comment:
Nit: Might not be the same behaviour that you want (e.g. if people navigate from other sites other than `/posts`) but there's a `goBack()` function in `history` that return you to previous page.

https://reacttraining.com/react-router/web/api/history